File: CASE_Simple_Example_V4/Hardened/HAMR/hamr/ext-c/FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner/FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner.c
#include <FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_api.h>
#include <FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner.c", "", "HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_initialise_", 0);

  // examples of api setter and logging usage

  DeclNewHAMR_SW_Mission(t0);
  HAMR_SW_Mission_example(SF &t0);
  api_put_FlightPlan__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF &t0);

  api_logInfo__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF string("Example logInfo"));

  api_logDebug__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF string("Example logDebug"));

  api_logError__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF string("Example logError"));
}

Unit HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner.c", "", "HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_finalise_", 0);
}

Unit HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_handle_MissionCommand_(
  STACK_FRAME
  HAMR_SW_RF_Msg_Impl value) {
  DeclNewStackFrame(caller, "FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner.c", "", "HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_handle_MissionCommand_", 0);

  DeclNewString(MissionCommandString);
  String__append(SF (String) &MissionCommandString, string("HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner_handle_MissionCommand called"));
  api_logInfo__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner (SF (String) &MissionCommandString);

  DeclNewString(_str);
  String__append(SF (String) &_str, string("Received on MissionCommand: "));
  HAMR_SW_RF_Msg_Impl_string_(SF (String) &_str, value);
  api_logInfo__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF (String) &_str);

  // examples of api getter usage

  DeclNewHAMR_SW_RF_Msg_Impl(t0);
  if(api_get_MissionCommand__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF &t0)) {
    DeclNewString(MissionCommand_str);
    String__append(SF (String) &MissionCommand_str, string("Received on MissionCommand: "));
    HAMR_SW_RF_Msg_Impl_string_(SF (String) &MissionCommand_str, &t0);
    api_logInfo__HAMR_SW_FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner(SF (String) &MissionCommand_str);
  }
}

File: CASE_Simple_Example_V4/Hardened/HAMR/src/main/component/HAMR/SW/FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner.scala
// #Sireum

package HAMR.SW

import org.sireum._
import HAMR._

// This file will not be overwritten so is safe to edit
object FlightPlanner_Impl_SW_FlightPlanner_FlightPlanner {

  def initialise(api: FlightPlanner_Impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.put_FlightPlan(SW.Mission.example())
  }

  def handle_MissionCommand(api: FlightPlanner_Impl_Operational_Api, value : SW.RF_Msg_Impl): Unit = {
    api.logInfo("example handle_MissionCommand implementation")
    api.logInfo(s"received ${value}")
    // example api usage

    val apiUsage_MissionCommand: Option[SW.RF_Msg_Impl] = api.get_MissionCommand()
    api.logInfo(s"Received on MissionCommand: ${apiUsage_MissionCommand}")
  }

  def activate(api: FlightPlanner_Impl_Operational_Api): Unit = { }

  def deactivate(api: FlightPlanner_Impl_Operational_Api): Unit = { }

  def finalise(api: FlightPlanner_Impl_Operational_Api): Unit = { }

  def recover(api: FlightPlanner_Impl_Operational_Api): Unit = { }
}
